### ISSUE TITLE
Unable to run new iptsd scripts on Android shell

### ISSUE THREAD

**Issue by hmtheboy154** [linux-surface/iptsd#184]
It seems like both [iptsd-find-hidraw](https://github.com/linux-surface/iptsd/blob/master/etc/scripts/iptsd-find-hidraw) and [iptsd-foreach](https://github.com/linux-surface/iptsd/blob/master/etc/scripts/iptsd-foreach) can't be able to executed on Android anymore, causing our init script unable to start iptsd. I checked and it seems like the script now uses `bash` and ultilize some bash features. So I tried 2 method so far:
- Force it to use Android's `sh`, this resulted in `sh` unable to find `BASH_SOURCE[0]` 
```
127|Standard PC (Q35 + ICH9, 2009):/ # sh /vendor/bin/iptsd-find-hidraw                                        
/vendor/bin/iptsd-find-hidraw[17]: BASH_SOURCE[0]: parameter not set
```
- Build bash on Android and then try to run the script. LineageOS has a fork of [bash](https://github.com/LineageOS/android_external_bash) that is buildable on Android, I tried to edit shebang and execute the script and got this output
```
1|Standard PC (Q35 + ICH9, 2009):/ # iptsd-find-hidraw                                                         
/vendor/bin/iptsd-find-hidraw: line 9: //iptsd-foreach: No such file or directory
```

**Comment by itmswt** [comment]
How have you gone with this one mate?
Can anyone also help so tge touch screen issue gets fixed?

**Comment by horvathcsabalaszlo** [comment]
I could run the IDC script via ADB root shell, but i don't know it's OK or not; at least it did not solve the "touch acts as mouse" problem :S

**Comment by hmtheboy154** [comment]
> I could run the IDC script via ADB root shell, but i don't know it's OK or not; at least it did not solve the "touch acts as mouse" problem :S

you'll only have single touch, not multi-touch. Try to touch more than 1 fingers

**Comment by horvathcsabalaszlo** [comment]
> > I could run the IDC script via ADB root shell, but i don't know it's OK or not; at least it did not solve the "touch acts as mouse" problem :S
> 
> you'll only have single touch, not multi-touch. Try to touch more than 1 finger

Yes, i'll be happy with it as the touchscreen is not working good on my device in multi mode; but acting as a mouse is a bigger problem. 
Also, the IDC did not solve the issue that the screen functions as a mouse, not as a touch device :D

**Comment by hmtheboy154** [comment]
this issue is fixed, @shadichy helped me converting these scripts to mksh

**Comment by horvathcsabalaszlo** [comment]
Thank you! Finally :)

**Comment by epicshepich** [comment]
I just installed Bliss OS v16.9.7 on my Surface Pro 7 _today_. What a coincidence! Is there a patch/workaround that I can apply, or will I have to wait for a new build?

**Comment by shadichy** [comment]
unfortunately, you'll have to wait
